perllib/FixMyStreet/Script/Merton/SendWaste.pm

```diff
@@ -55,6 +57,27 @@ sub _problems {
     });
 }
 
+sub set_echo_id {
+    my ($self, $problem) = @_;
+
+    unless ($problem->get_extra_field_value('echo_id')) {
+        my $cobrand = $problem->get_cobrand_logged;
+        my $cfg = $cobrand->feature('echo');
+        my $echo = Integrations::Echo->new(%$cfg);
+        my $event = $echo->GetEvent($problem->external_id);
+        return unless $event && $event->{Id};
+
+        my $row2 = FixMyStreet::DB->resultset('Problem')->search({ id => $problem->id }, { for => \'UPDATE' })->single;
+        $row2->update_extra_field({
+            name => 'echo_id',
+            value => $event->{Id},
+        });
+        $row2->update;
```

Comment:
Guess we want a problem->discard_changes here, so it picks up the data?

Reply:
Yes that has done the trick. I thought for some reason that it wouldn't be a good idea to change the existing `$problem`, perhaps because of 57c8a7bee25e570215ed55252a972d3bff0316b7?

Reply:
Think it's fine, we're not in a transaction or anything, and it's before any bit of the loop anyway. I'd want it at the end, not the beginning, though?